packages/orchestration/src/exos/cosmos-orchestration-account.js

```diff
@@ -138,20 +139,18 @@ export const prepareCosmosOrchestrationAccountKit = (
       }),
       holder: IcaAccountHolderI,
       invitationMakers: M.interface('invitationMakers', {
-        Delegate: M.callWhen(ChainAddressShape, AmountArgShape).returns(
-          InvitationShape,
+        Delegate: M.call(ChainAddressShape, AmountArgShape).returns(
+          M.promise(),
```

Comment:
For the `invitationMakers` interfaces, the return promise unwraps to an `InvitationShape`, not a `VowShape`. The offerHandler of the Invitation is what needs to be a Vow.

Reply:
> shouldn't this be explicitly a promise for void?

That's not expressible as a pattern. It's not pass-invariant.

Reply:
(Nit: in typescript `void` does not signify `undefined` but rather a kind of don't care. I'll ignore that below.)

In is indeed not expressible as a pattern, but in this case not because of pass invariance. Patterns (as opposed to guards) can only enforce what they can immediately check, and a pattern cannot immediately check what the promise will eventually fulfill to. 

The original, by using an `M.callWhen` method guard with a `returns(InvitationShape)` does enforce that `Delegate` returns a promise for something that matches `InvitationShape`, because it uses the `callWhen` delay to validate the promise's fulfillment before passing it through. Likewise, within an `M.callWhen` you can say `M.await(FooShape)` as a parameter guard. If the argument is a promise, this will enforce that the promise fulfills to something that matches `FooShape` before passing that fulfillment along as the argument to the raw method.